src/Desktop/UIAutomation/A11yAutomation.cs

```diff
@@ -391,23 +393,13 @@ public static A11yElement NormalizedElementFromPoint(int xPos, int yPos, TreeVie
             return null;
         }
 
-        /// <summary>
-        /// Get Property programmatic Name from UIA service.
-        /// </summary>
-        /// <param name="pid"></param>
-        /// <returns></returns>
-        public static string GetPropertyProgrammaticName(int pid)
-        {
-            return UIAutomation.GetPropertyProgrammaticName(pid);
-        }
-
-        /// <summary>
-        /// Get DesktopElement from Desktop
-        /// </summary>
-        /// <returns></returns>
-        public static DesktopElement ElementFromDesktop()
+        private void EnsureContextConsistency(DesktopDataContext dataContext)
         {
-            return new DesktopElement(UIAutomation.GetRootElement());
+            if (dataContext == null) throw new ArgumentNullException(nameof(dataContext));
+            if (!ReferenceEquals(dataContext.A11yAutomation, this))
```

Comment:
I don't love that we have to make this check, can we possibly keep the methods where we pass in the dataContext static and only use the UIAutomation instance kept in the data context?

Reply:
I tried reverting the classes that receive the `DesktopDataContext` object to static, and it wasn't pretty. We had static methods calling non-static ones and the code got really hard to read. I can investigate splitting this into 2 classes, but we may have the same problem. Given the size of the current PR, I'd be inclined to move that effort into a separate PR

Reply:
I agree, following up in a different PR makes sense :)